### ISSUE TITLE
Unable to change Wink device properties after new schema changes

### ISSUE THREAD

**Issue by busadave13** [openT2T/translators#62]
After integrating with the new schema changes I can get the Wink device properties from the translator with getThermostatResURI but i'm unable to change the device properties using postThermostatResURI . The post succeeds but the device does not update.

**Comment by tjaffri** [comment]
Adding @begillil and @arjun-msft who are also both running into this.

I looked at the translator in the master branch, and ran the tests there. It appears the issue is that we need to first set the mode before we can get/set the heating/cooling setpoints.

So this is a test issue. In the test we need to do something like:

1. Set the mode to "heat_only"
2. Now set the heating setpoint, i.e. min_set_point
3. Now read the heating setpoint (and this should work now)

So the above should fix the immediate issue you guys are running into as long as somewhere in the flow we set the mode. To do this, we need to:

1. Add the mode to the schema
2. Implement getting/setting mode in the translator,
3. Update associated unit tests to set the mode prior to querying relevant setpoint

There is a broader issue however... which we need to discuss. Any "real" thermostat will have different available modes... e.g. in my house I might have a heater and an AC so my nest supports both heat and cool modes. However, in some other house they might only have a heater so they are in heat_only mode. Per the Wink docs http://docs.wink.apiary.io/#reference/device/thermostats they say for the mode desired state field that:

> One of ["cool_only", "heat_only", "auto", "aux"], allowed value depends in mode.choices in capabilities

So we need to query "capabilities" to see what this particular thermostat supports and then auto-set the mode to that (or otherwise figure out how we can get the user's desired mode value)

Hope this helps unblock you in the short term, though we need to keep discussing for the longer term so we are not disabling people's desired modes.

P.S. It looks like the test was broken in the master branch as well because it was setting mode to heat_only and then querying cooling set point. So I will send a PR shortly to fix the test in master, so we have that for reference.

**Comment by arjun-msft** [comment]
I did some investigation and the issue Taqi mentions is an issue but there is another issue which is when we are setting the putPayload we are adding in the data object which is not valid.  I verified this using fiddler and a local copy of this in the cloud sample.  For the case of auto which is the default for the simulator, we can set the min/max set point without having to change the mode first.

@busadave13 @begillil if you can verify this change fixes on your side that would be awesome.

**Comment by begillil** [comment]
@arjun-msft, I also found the same fix while I was working on my changes today.

**Comment by tjaffri** [comment]
Looks like the mode thing was a red herring, then.... thanks guys! I'm going to close this one.
